framework/src/main/java/org/checkerframework/framework/type/treeannotator/PropagationTreeAnnotator.java

```diff
@@ -141,8 +141,7 @@ public Void visitNewArray(NewArrayTree tree, AnnotatedTypeMirror type) {
 
                 // If the method accepts varargs, we handle the vararg parameter after this
                 // for-loop.
-                int loopBound = isVarargs ? parametersCount - 1 : parametersCount;
-                for (int i = 0; i < loopBound; i++) {
+                for (int i = 0; i < parametersCount; i++) {
                     @SuppressWarnings("interning") // Tree must be exactly the same.
                     boolean foundArgument = methodInvocationTree.getArguments().get(i) == tree;
```

Comment:
nit: maybe a comment saying this won't throw for varargs because parametersCount is already adapted.